piker/brokers/ib.py

```diff
@@ -1183,112 +1201,217 @@ def pack_position(pos: Position) -> Dict[str, Any]:
-            # from the execution details event
-            msg = {
-                'reqid': trade.order.orderId,
-                'status': status.status,
-                'filled': status.filled,
-                'reason': status.whyHeld,
+                    filled=status.filled,
+                    reason=status.whyHeld,
+                    # this seems to not be necessarily up to date in the
+                    # execDetails event.. so we have to send it here I guess?
+                    remaining=status.remaining,
 
-                # this seems to not be necessarily up to date in the
-                # execDetails event.. so we have to send it here I guess?
-                'remaining': status.remaining,
-            }
+                    broker_details={'name': 'ib'},
+                )
 
-        elif event_name == 'fill':
+            elif event_name == 'fill':
 
-            trade, fill = item
-            execu = fill.execution
+                # for wtv reason this is a separate event type
+                # from IB, not sure why it's needed other then for extra
+                # complexity and over-engineering :eyeroll:.
+                # we may just end up dropping these events (or
+                # translating them to ``Status`` msgs) if we can
+                # show the equivalent status events are no more latent.
 
-            msg = {
-                'reqid': execu.orderId,
-                'execid': execu.execId,
+                # unpack ib_insync types
+                # pep-0526 style:
+                # https://www.python.org/dev/peps/pep-0526/#global-and-local-variable-annotations
+                trade: Trade
+                fill: Fill
+                trade, fill = item
+                execu: Execution = fill.execution
+
+                # TODO: normalize out commissions details?
+                details = {
+                    'contract': asdict(fill.contract),
+                    'execution': asdict(fill.execution),
+                    'commissions': asdict(fill.commissionReport),
+                }
 
                 # supposedly IB server fill time
-                'broker_time': execu.time,  # converted to float by us
-                # ns from main TCP handler by us inside ``ib_insync`` override
-                'time': fill.time,
-                'time_ns': time.time_ns(),  # cuz why not
-                'action': action_map[execu.side],
-                'size': execu.shares,
-                'price': execu.price,
-            }
-
-        elif event_name == 'error':
-            msg = item
-
-            # f$#$% gawd dammit insync..
-            con = msg['contract']
-            if isinstance(con, Contract):
-                msg['contract'] = asdict(con)
-
-            if msg['reqid'] == -1:
-                log.error(pformat(msg))
-
-            # don't forward, it's pointless..
-            continue
+                details['broker_time'] = execu.time
```

Comment:
probably do do these in the static declaration above, and maybe further do it all in one line below in the message creation line?